trin-core/src/portalnet/protocol.rs

```diff
@@ -97,6 +123,54 @@ impl JsonRpcHandler {
                         .collect();
                     let _ = cmd.resp.send(Ok(Value::Array(routing_table_info)));
                 }
+                DummyHistoryNetworkData => {
```

Comment:
@carver Any thoughts on this? Imo it's gross to read - but it does handle all the error cases correctly. It wasn't obvious to me how to break it out into a simpler function - but I'm happy to spend some more time cleaning it up.

Reply:
Yeah, at the very least, I'd probably put all of this in its own method. I think it will be more obvious how to generalize as we add more requests that go to the same subnet. (But it does seem like passing in just a couple variables could mean being able to put most of this into a single shared method like `proxy_query_to_subnet(subnet_tx, msg_kind) -> response`.